service/cloudfront/sign/sign_cookie.go

```diff
@@ -0,0 +1,141 @@
+	keyID   string
+	privKey *rsa.PrivateKey
+}
+
+//CookieOptions optional settings for Cookie Definition, these can also be set manually
+type CookieOptions struct {
+	Path   string
+	Domain string
+	Secure bool
+}
+
+//Example:
+
+// func handler(w http.ResponseWriter, r *http.Request) {
+
+//  // sign cookie to be valid for 30 minutes from now, expires one hour from now, and
+//  // restricted to the 192.0.2.0/24 IP address range.
+
+//  //http://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/private-content-setting-signed-cookie-custom-policy.html
+// 	p := &sign.Policy{
+// 		Statements: []sign.Statement{
+// 			{
+//				Resource: RawCloudFrontURL, //read the provided documentation on how to set this correctly, you'll probably want to use wildcards
+// 				Condition: sign.Condition{
+// 					// Optional IP source address range
+// 					IPAddress: &sign.IPAddress{SourceIP: "192.0.2.0/24"},
+// 					// Optional date URL is not valid until
+// 					DateGreaterThan: &sign.AWSEpochTime{time.Now().Add(30 * time.Minute)},
+// 					// Required date the URL will expire after
+// 					DateLessThan: &sign.AWSEpochTime{time.Now().Add(1 * time.Hour)},
+// 				},
+// 			},
+// 		},
+// 	}
+
+// 	//load your private key and convert it to type rsa.PrivateKey
+// 	privKey, err := sign.LoadPEMPrivKeyFile("privatekey.pem")
+// 	if err != nil {
+// 		fmt.Println("error", err)
+// 	}
+
+// 	//key ID that represents the key pair associated with the private key
+// 	keyID := "WOIERULSDLKEWRIU"
+
+// 	//set credentials to the cookiesigner
+// 	signer := sign.NewCookieSigner(keyID, privKey)
+
+// 	//creates 3 signed cookies, provide an optional Cookie Options struct to specify other options
+// 	o := &sign.CookieOptions{
+// 		Path:   "/",
+// 		Domain: ".cNameAssociatedWithMyDistribution.com",
+// 		Secure: true, //make sure your app/site can handle https payloads, otherwise set this to false
+// 	}
+// 	//http://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/private-content-setting-signed-cookie-custom-policy.html#private-content-custom-policy-statement-signed-cookies-examples
+//  //http://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/CNAMEs.html
+// 	//avoid adding an Expire or MaxAge. See provided AWS Documentation for more info
+
+// 	policy, signature, key, err := signer.SignCookies(p)
+// 	if err != nil {
+// 		fmt.Println("error", err)
+// 	}
+
+// 	http.SetCookie(w, policy)
+// 	http.SetCookie(w, signature)
+// 	http.SetCookie(w, key)
+
+// }
+
+// NewCookieSigner constructs and returns a new CookieSigner to be used to for signing
+// Amazon CloudFront URL resources with.
+func NewCookieSigner(keyID string, privKey *rsa.PrivateKey) *CookieSigner {
+	return &CookieSigner{
+		keyID:   keyID,
+		privKey: privKey,
+	}
+}
+
+//SignCookies CookieOptions are optional, if unused, just pass in nil
+func (c CookieSigner) SignCookies(p *Policy, o *CookieOptions) (cPolicy, cSignature, cKey *http.Cookie, err error) {
```

Comment:
This probably should follow the `URLSigner`'s pattern and be `SignWithPolicy`.

In addition what do you think about adding a `Sign` method similar to `URLSigner`'s that takes a `resource` string and a `expires` time.Time instead of a Policy? The `Sign` method could then use the `NewCannedPolicy` func to created a policy from a resource and expires values.